### PR TITLE
Add Lambda rule

### PR DESCRIPTION
**Pull request by Morganamilo**

### CODE REVIEW

**Review by graham-russell — COMMENTED** [review]
Could you generate provider docs with tfplugindocs (https://github.com/hashicorp/terraform-plugin-docs). Once installed, navigate to repository root and run `tfplugindocs`

Please also create examples/playground/rule_aws_lambda.tf & examples/resources/rule_aws_lambda.tf

**Comment by Morganamilo** [comment]
Whoops I somehow totally forgot the examples

File: examples/resources/rule_aws_sqs.tf [hunk 1]
```diff
@@ -5,15 +5,15 @@ resource "ably_rule_sqs" "rule0" {
...
-    mode     = "assumeRole",
-    role_arn = "cccc"
-  }
   target = {
     region         = "us-west-1",
     aws_account_id = "123456789012",
     queue_name     = "aaaaaa",
     enveloped      = false,
     format         = "json"
+    aws_authentication = {
```
Comment (graham-russell) [inline examples/resources/rule_aws_sqs.tf:14]: This one should be `authentication` rather than `aws_authentication`

Comment (Morganamilo) [reply examples/resources/rule_aws_sqs.tf]: Thanks. Had to miss one.

**Review by graham-russell — APPROVED** [review]
LGTM

**Pull request merged**